pkg/pipelinerunmetrics/metrics.go

```diff
@@ -396,9 +414,37 @@ func (r *Recorder) RunningPipelineRuns(lister listers.PipelineRunLister) error {
 	var runningPipelineRuns int
 	var trsWaitResolvingTaskRef int
 	var prsWaitResolvingPipelineRef int
+	countMap := map[string]int{}
 
 	for _, pr := range prs {
+		pipelineName := getPipelineTagName(pr)
+		pipelineRunKey := ""
+		mutators := []tag.Mutator{
+			tag.Insert(namespaceTag, pr.Namespace),
+			tag.Insert(pipelineTag, pipelineName),
+		}
+		if r.cfg != nil {
```

Comment:
The `NewRecorder` function uses config to initialise views in the `viewRegister` function, but it does not store config as part of the recorder itself. This means that `r.cfg` here is always going to be `nil`.

This results in having the views set up correctly but the counting being always at cluster level.
Existing testing is unfortunately insufficient to see the issue, as the test case only sets up a single running PR which does not allow for any aggregation or filtering to happen.

I tried the following change to the test:

```golang
	for _, pipelineRun := range []*v1.PipelineRun{
		newPipelineRun(corev1.ConditionTrue, "testns"),
		newPipelineRun(corev1.ConditionUnknown, "testns"),
		newPipelineRun(corev1.ConditionUnknown, "testns2"),
		newPipelineRun(corev1.ConditionFalse, "testns"),
	} {
		if err := informer.Informer().GetIndexer().Add(pipelineRun); err != nil {
			t.Fatalf("Adding TaskRun to informer: %v", err)
		}
	}

	ctx = getConfigContextRunningPRLevel("namespace")
	recorder, err := NewRecorder(ctx)
	if err != nil {
		t.Fatalf("NewRecorder: %v", err)
	}

	if err := recorder.RunningPipelineRuns(informer.Lister()); err != nil {
		t.Errorf("RunningPipelineRuns: %v", err)
	}
	metricstest.CheckLastValueData(t, "running_pipelineruns", map[string]string{"namespace": "testns"}, 1)
	metricstest.CheckLastValueData(t, "running_pipelineruns", map[string]string{"namespace": "testns2"}, 1)
```

which results in both pipelineruns being counted in each namespace:

```
    /System/Volumes/Data/go/src/github.com/tektoncd/pipeline/pkg/pipelinerunmetrics/metrics_test.go:563: Reporter expected a different tag value for key metric running_pipelineruns key namespace got testns2 want testns
    /System/Volumes/Data/go/src/github.com/tektoncd/pipeline/pkg/pipelinerunmetrics/metrics_test.go:563: Reporter.Report() wrong value metric running_pipelineruns got 2 want 1
    /System/Volumes/Data/go/src/github.com/tektoncd/pipeline/pkg/pipelinerunmetrics/metrics_test.go:564: Reporter.Report() wrong value metric running_pipelineruns got 2 want 1
```